Jenkinsfile

```diff
@@ -23,17 +43,80 @@ node('us-east-1 && ubuntu && docker && !gpu') {
                     ./mach package
                 '''
             }
-            stage('Upload APK') {
-                archiveArtifacts allowEmptyArchive: true, artifacts: 'mozilla-release/objdir-frontend-android/dist/*.apk'
+        }
+        withEnv(["INSTANCE_ID=${instance_id}"]){
+            timeout(5){
+                stage('Genymotion Status'){
+                    def status = sh(returnStdout: true, script: """
+                            aws ec2 describe-instance-status --region=us-east-1 --instance-id $INSTANCE_ID --query 'InstanceStatuses[].InstanceStatus[].Details[].Status' --output text
+                        """).trim()
+                    while (status != 'passed') {
```

Comment:
is there no better way?

Reply:
Other methods are flaky, and has a failure rate of 30-50%. This way, it is working 100%.